daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/FNFunctions.scala

```diff
@@ -803,6 +803,52 @@ case class FNFloor(recipe: CompiledDPath, argType: NodeInfo.Kind) extends FNOneA
   }
 }
 
+case class DFDLXLeftShift(recipes: List[CompiledDPath],argType: NodeInfo.Kind) extends FNTwoArgs(recipes) {
+  override def computeValue(arg1: DataValuePrimitive, arg2: DataValuePrimitive, dstate: DState): DataValuePrimitive = {
+
+    val shift = asInt(arg2.getAnyRef).intValue
+    argType match {
+      case NodeInfo.Long => {
+        val leftShift = asLong(arg1).longValue()
+        if (shift >=64) {
+          0
+        }
+        else {
+          leftShift << shift
+        }
+      }
+      case NodeInfo.Int => {
+        val leftShift = asInt(arg1.getAnyRef).intValue()
+        if (shift >=32) {
+           0
+        }
+        else {
+           leftShift << shift
+        }
+      }
+      case NodeInfo.Short => {
+        val leftShift = asShort(arg1.getAnyRef).shortValue()
+        if (shift >=16) {
+           0
+        }
+        else {
+           leftShift << shift
+        }
+      }
+      case NodeInfo.Byte => {
+        val leftShift = asByte(arg1.getAnyRef).byteValue()
+        if (shift >=8) {
+           0
+        }
+        else {
+           leftShift << shift
+        }
+      }
+      case _ => Assert.invariantFailed(String.format("Left shift not supported"))
```

Comment:
The String.format isn't neded here. Note a big deal, but cleans things up a little bit. Also, because this line is impossible to test, we need to add an annotation to make sure code coverage doesn't ding us on this line. So this probably just wants to be some thing like

```scala
// $COVERAGE-OFF$
case _ => Assert.invariantFailed("Left shift for type should be impossible")
// $COVERAGE-ON#
```